### ISSUE TITLE
[no-let] Add option to allow for ignoring `let`s in for loop initializers.

### ISSUE THREAD

**Issue by RebeccaStevens** [jonaskello/eslint-plugin-functional#270]
```ts
for (let i = 0; ; ) {
  // ^^^ ignore this let
}
```

The `no-loop` rule can handle preventing loops.

**Comment by RebeccaStevens** [comment]
I'm not sure whether this should be added or not. There are already options that can be used to handle this:

```
/* eslint functional/no-let: ["error", "allowLocalMutation": true] */
for (let i = 0; ; ) {}
```

```
/* eslint functional/no-let: ["error", "ignorePattern": "^mutable"] */
for (let mutableI = 0; ; ) {}
```

But then again, this option could still be a nice addition. @jonaskello Thoughts?

**Comment by schoening** [comment]
Entire point of this package is to do more FP. FP has maps and recursion and avoids imperative loops. I think you should add a eslint comment where needed and start using a functional approach where possible:
```
import { map } from 'ramda';
```

**Comment by jonaskello** [comment]
@schoening Yes that is the point :-). You can already disallow imperative loops entirely by using the [no-loop](https://github.com/jonaskello/eslint-plugin-functional/blob/master/docs/rules/no-loop-statement.md) rule. IIRC one problem with doing recursion for everything in javascript is that tail call optimization is not implemented (but It think it is included in the standard for ECMAScript). The alternative is to use external libraries but in some cases that will hurt performance. If you have very performance sensitive code it could make sense to allow imperative loops.

@RebeccaStevens I agree that allowing for-loops but not allowing let is a strange combination because for-loops does not make sense without let. I would probably use the mutable prefix to solve this like you documented above. However the option you propose still makes sense to me. I think it is correct to have it as an option and not the default (as proposed). The expectation if you just enable no-let would be that no `let` are allowed anywhere.
